src/backend/distributed/metadata/metadata_cache.c

```diff
@@ -2185,7 +2186,19 @@ HasOverlappingShardInterval(ShardInterval **shardIntervalArray,
 bool
 CitusHasBeenLoaded(void)
 {
-	if (!MetadataCache.extensionLoaded || creating_extension)
+	/*
+	 * MetadataCache.extensionLoaded and MetadataCache.extensionNotLoaded
+	 * cannot be true at the same time.
+	 */
+	Assert(!(MetadataCache.extensionLoaded && MetadataCache.extensionNotLoaded));
```

Comment:
nit: I think most of these checks become easier to understand with a 3 element `enum` instead of 2 booleans. Something like this: `EXTENSION_LOADED_UNKNOWN` `EXTENSION_LOADED` `EXTENSION_NOT_LOADED` (feel free to choose better names)